osu.Game.Rulesets.Osu/Replays/OsuAutoGenerator.cs

```diff
@@ -339,6 +339,10 @@ private void addHitObjectClickFrames(OsuHitObject h, Vector2 startPosition, floa
 
             AddFrameToReplay(startFrame);
 
+            // ~477 as per stable.
+            const float spin_rpm = 60000f / 20 * (180 / MathF.PI) / 360;
```

Comment:
This intentionally doesn't use `MathUtils` so it can remain a `const`. This allows hovering the variable to see the current value, which I quite like:

![CleanShot 2023-11-17 at 07 46 58](https://github.com/ppy/osu/assets/191335/8229e444-65f5-4257-816a-9829e4a69b4e)

Reply:
It may be just me but I find this pretty inscrutable? What's the 20? Why 180 over pi and then 360 again?

What stable does is

```csharp
            rpm = rpm * decay + (1.0 - decay) * (Math.Abs(velocityCurrent) * 1000) / (Math.PI * 2) * 60;
```

which I'd either copy verbatim, or rearrange some more to

```
(0.05 / (2 * pi)) * 60000
```

which is just converting 0.05 rad/ms to rotations per minute.

Reply:
I just assumed we were already copying stable so tried to make it work 😅. What you have seems better for sure.